tests/std/tests/P0896R4_ranges_algorithm_machinery/test.cpp

```diff
@@ -983,3 +984,128 @@ namespace gh_1089 {
+        int operator*() const requires (I == iterator_status::not_lvalue_reference);
+        int& operator*() const requires (I != iterator_status::not_lvalue_reference);
+
+        iterator_archetype& operator++();
+        void operator++(int) requires (I != iterator_status::forward);
+        iterator_archetype operator++(int) requires (I == iterator_status::forward);
+
+        bool operator==(std::default_sentinel_t) const;
+        bool operator==(iterator_archetype const&) const requires (I == iterator_status::forward);
+        // clang-format on
+    };
+    // Verify iterator_archetype
+    STATIC_ASSERT(!input_iterator<iterator_archetype<iterator_status::not_input>>);
+    STATIC_ASSERT(input_iterator<iterator_archetype<iterator_status::not_lvalue_reference>>);
+    STATIC_ASSERT(input_iterator<iterator_archetype<iterator_status::different_reference_and_value>>);
+    STATIC_ASSERT(input_iterator<iterator_archetype<iterator_status::input>>);
+    STATIC_ASSERT(!forward_iterator<iterator_archetype<iterator_status::input>>);
+    STATIC_ASSERT(forward_iterator<iterator_archetype<iterator_status::forward>>);
+
+    template <class I>
+    inline constexpr bool has_lvalue_reference = std::is_lvalue_reference_v<std::iter_reference_t<I>>;
+    STATIC_ASSERT(has_lvalue_reference<iterator_archetype<iterator_status::not_input>>);
+    STATIC_ASSERT(!has_lvalue_reference<iterator_archetype<iterator_status::not_lvalue_reference>>);
+    STATIC_ASSERT(has_lvalue_reference<iterator_archetype<iterator_status::different_reference_and_value>>);
+    STATIC_ASSERT(has_lvalue_reference<iterator_archetype<iterator_status::input>>);
+    STATIC_ASSERT(has_lvalue_reference<iterator_archetype<iterator_status::forward>>);
+
+    template <class I>
+    inline constexpr bool same_reference_value =
+        std::same_as<std::remove_cvref_t<std::iter_reference_t<I>>, std::iter_value_t<I>>;
+    STATIC_ASSERT(same_reference_value<iterator_archetype<iterator_status::not_input>>);
+    STATIC_ASSERT(same_reference_value<iterator_archetype<iterator_status::not_lvalue_reference>>);
+    STATIC_ASSERT(!same_reference_value<iterator_archetype<iterator_status::different_reference_and_value>>);
+    STATIC_ASSERT(same_reference_value<iterator_archetype<iterator_status::input>>);
+    STATIC_ASSERT(same_reference_value<iterator_archetype<iterator_status::forward>>);
+
+    // Validate _No_throw_input_iterator
+    STATIC_ASSERT(!_No_throw_input_iterator<iterator_archetype<iterator_status::not_input>>);
+    STATIC_ASSERT(!_No_throw_input_iterator<iterator_archetype<iterator_status::not_lvalue_reference>>);
+    STATIC_ASSERT(!_No_throw_input_iterator<iterator_archetype<iterator_status::different_reference_and_value>>);
+    STATIC_ASSERT(_No_throw_input_iterator<iterator_archetype<iterator_status::input>>);
+    STATIC_ASSERT(_No_throw_input_iterator<iterator_archetype<iterator_status::forward>>);
+
+    enum class sentinel_status : int { not_sentinel, good };

```

Comment:
I have a mild preference for symmetric(ish) naming here: `{not_sentinel, is_sentinel}` or `{bad, good}`.  Perhaps even `{no, yes}` as I think that would read nicely ("`sentinel_status::no`", "`sentinel_status::yes`").  
Feel free to ignore.

Reply:
`{ no, yes }` is fine here.